### PR TITLE
os, mgr, librbd, client: cleanups related to 32-bit platforms

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by batrick — APPROVED** [review]

File: src/librbd/deep_copy/ObjectCopyRequest.cc [hunk 1]
```diff
@@ -614,7 +614,8 @@ void ObjectCopyRequest<I>::merge_write_ops() {
 
         m_snapshot_sparse_bufferlist[src_snap_seq].insert(
           object_extent.offset, object_extent.length,
-          {io::SPARSE_EXTENT_STATE_DATA, object_extent.length,\
+          {io::SPARSE_EXTENT_STATE_DATA,
+           static_cast<size_t>(object_extent.length),
```
Comment (idryomov) [inline src/librbd/deep_copy/ObjectCopyRequest.cc:618]: I think we should change SparseExtent::length to uint64_t instead.  Not sure why it's size_t -- ReadExtent::length is uint64_t and related length() methods in the surrounding classes all return uint64_t.

**Review by idryomov — CHANGES_REQUESTED** [review]

File: src/librbd/deep_copy/ObjectCopyRequest.cc (hunk 1 above)
Comment (tchaikov) [reply src/librbd/deep_copy/ObjectCopyRequest.cc]: @idryomov thanks. that makes much more sense. updated and repushed.

**Review by idryomov — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-05-12_04:48:43-rados-wip-kefu-testing2-2021-05-12-1103-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-05-13_03:34:40-rados-wip-kefu-testing-2021-05-12-2313-distro-basic-smithi/

**Review by ifed01 — APPROVED** [review]
Allocator's changes are OK

**Review by neha-ojha — APPROVED** [review]

**Comment by batrick** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2021-May-11

**Pull request merged**